delete-user-data/functions/src/index.ts

```diff
@@ -117,12 +118,24 @@ const clearFirestoreData = async (firestorePaths: string, uid: string) => {
   const paths = extractUserPaths(firestorePaths, uid);
   const promises = paths.map(async (path) => {
     try {
-      logs.firestorePathDeleting(path);
-      await admin
-        .firestore()
-        .doc(path)
-        .delete();
-      logs.firestorePathDeleted(path);
+      const isRecursive = config.firestoreDeleteMode === 'recursive';
+
+      if (!isRecursive) {
+        logs.firestorePathDeleting(path, false);
+        await admin
+          .firestore()
+          .doc(path)
+          .delete();
+        logs.firestorePathDeleted(path, false);
+      } else {
+        logs.firestorePathDeleting(path, true);
+        await firebase_tools.firestore.delete(path, {
+          project: process.env.GCLOUD_PROJECT,
```

Comment:
use process.env.PROJECT_ID, since we set that env variable when deploying an extension so we know it's guaranteed to be there. process.env.GCLOUD_PROJECT is not reliable.